client/components/app.jsx

```diff
@@ -21,14 +22,21 @@ class App extends React.Component {
     if (name === 'catalog') {
       this.setState({
         view: {
-          name: 'catalog',
+          name: name,
           params: {}
         }
       });
-    } else {
+    } else if (name === 'cart') {
       this.setState({
         view: {
           name: name,
+          params: {}
+        }
+      });
+    } else {
+      this.setState({
+        view: {
```

Comment:
You should be able to get rid of these `if/else` statements in `setView`. This method is told by its caller what the view name should be `showView('catalog, dammit', {})`.